### PR TITLE
chore(webkit): always attach to all pages, simplify initialization

### PR DESCRIPTION
**Pull request by dgozman**

### CODE REVIEW

File: src/webkit/wkBrowser.ts [hunk 1]
```diff
@@ -108,27 +110,17 @@ export class WKBrowser extends platform.EventEmitter implements Browser {
...
+      return;
     if (!context)
       context =  this._defaultContext;
     const pageProxySession = new WKSession(this._connection, pageProxyId, `The page has been closed.`, (message: any) => {
       this._connection.rawSend({ ...message, pageProxyId });
     });
-    const pageProxy = new WKPageProxy(pageProxySession, context, () => {
-      if (!pageProxyInfo.openerId)
-        return null;
-      const opener = this._pageProxies.get(pageProxyInfo.openerId);
```
Comment (yury-s) [inline src/webkit/wkBrowser.ts]: The idea was to not create a strong reference to the opener but it should not be a big issue I think.

File: src/webkit/wkBrowser.ts [hunk 2]
```diff
@@ -137,19 +129,25 @@ export class WKBrowser extends platform.EventEmitter implements Browser {
 
   _onPageProxyDestroyed(event: Protocol.Browser.pageProxyDestroyedPayload) {
     const pageProxyId = event.pageProxyId;
-    const pageProxy = this._pageProxies.get(pageProxyId)!;
+    const pageProxy = this._pageProxies.get(pageProxyId);
+    if (!pageProxy)
```
Comment (yury-s) [inline src/webkit/wkBrowser.ts]: How can this happen?

File: src/webkit/wkPageProxy.ts [hunk 3]
```diff
@@ -28,30 +27,34 @@ const isPovisionalSymbol = Symbol('isPovisional');
...
   private readonly _pageProxySession: WKSession;
   readonly _browserContext: BrowserContext;
-  private readonly _openerResolver: () => WKPageProxy | null;
-  private _pagePromise: Promise<Page> | null = null;
-  private _wkPage: WKPage | null = null;
-  private readonly _firstTargetPromise: Promise<void>;
-  private _firstTargetCallback?: () => void;
-  private _pagePausedOnStart: boolean = false;
+  private readonly _opener: WKPageProxy | null;
+  private _pagePromise: Promise<Page | null>;
```
Comment (yury-s) [inline src/webkit/wkPageProxy.ts]: nit: readonly

File: src/webkit/wkPageProxy.ts [hunk 4]
```diff
@@ -28,30 +27,34 @@ const isPovisionalSymbol = Symbol('isPovisional');
...
+  private _pagePromiseFulfill: (page: Page | null) => void = () => {};
+  private _pagePromiseReject: (error: Error) => void = () => {};
+  private _wkPage: WKPage;
```
Comment (yury-s) [inline src/webkit/wkPageProxy.ts]: ditto

File: src/webkit/wkPageProxy.ts [hunk 5]
```diff
@@ -138,26 +109,38 @@ export class WKPageProxy {
...
+        await this._wkPage.initialize(session);
+        page = this._wkPage._page;
+      } catch (e) {
+        if (!this._pageProxySession.isDisposed())
+          error = e;
+      }
+      if (error)
+        this._pagePromiseReject(error);
+      else
+        this._pagePromiseFulfill(page!);
```
Comment (yury-s) [inline src/webkit/wkPageProxy.ts]: page could be undefined here if wkProxySession is disposed, should we change the type of the promise/callback?

**Review by yury-s — APPROVED** [review]

File: src/webkit/wkBrowser.ts (hunk 1 above)
Comment (dgozman) [reply src/webkit/wkBrowser.ts]: Yeah, I think it's fine.

File: src/webkit/wkBrowser.ts (hunk 2 above)
Comment (dgozman) [reply src/webkit/wkBrowser.ts]: If the proxy belongs to default context, and `!this._attachToDefaultContext`.

File: src/webkit/wkPageProxy.ts (hunk 3 above)
Comment (dgozman) [reply src/webkit/wkPageProxy.ts]: Done.

File: src/webkit/wkPageProxy.ts (hunk 4 above)
Comment (dgozman) [reply src/webkit/wkPageProxy.ts]: Done.

File: src/webkit/wkPageProxy.ts (hunk 5 above)
Comment (dgozman) [reply src/webkit/wkPageProxy.ts]: Good catch, updated.

**Pull request merged**